Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -3,28 +3,80 @@ import * as toolLib from 'vsts-task-tool-lib/tool';
+    private async installFromGlobalJson(osSuffixes: string[]) {
+        let filePathsToGlobalJson = this.getFiles(this.workingDirectory ? this.workingDirectory : ".", "global.json");
+        if(filePathsToGlobalJson.length == 0){
+            throw tl.loc("FailedToFindGlobalJson");
+        }
+        let sdkVersions = new Array<{ name: string, toolPath?: string }>();
+        // read all global files
+        for (let index = 0; index < filePathsToGlobalJson.length; index++) {
+            const filePath = filePathsToGlobalJson[index];
+            let globalJson = {} as { sdk: { version: string } };  
+            console.log("global.json found here: " + filePath);
+            try {
+                globalJson = (JSON.parse(fileSystem.readFileSync(filePath).toString())) as { sdk: { version: string } };
+            } catch (error) {
+                throw tl.error("FailedToReadGlobalJson \r\n" + 
+                "The global.json file isn't valid. (" + filePath + ") Please take a look in the documentation for global.json (https://docs.microsoft.com/en-us/dotnet/core/tools/global-json)");
+            }
+            if(globalJson == null || globalJson.sdk == null || globalJson.sdk.version == null){
+                throw tl.error("FailedToReadGlobalJson \r\n" +  
+                "The global.json file isn't valid. (" + filePath + ") Please take a look in the documentation for global.json (https://docs.microsoft.com/en-us/dotnet/core/tools/global-json)");
+            }
+            sdkVersions.push({ name: globalJson.sdk.version, toolPath: null });            
+        }
+        // check each version if it installed in the cache
+        for (let index = 0; index < sdkVersions.length; index++) {
+            const d = sdkVersions[index];
+            let path = this.getLocalTool(d.name);
+            d.toolPath = path;
+        }      
+        // download all sdk that are not in the cache
+        let sdksToDownload = sdkVersions
+            // remove all tools that are present in cache.
+            .filter(function (sdk) {return sdk.toolPath == null})
+            // distinct on version name
+            .filter(function (value, index, self) {
+                return self.map(function (sdk){return sdk.name}).indexOf(value.name) == index;
+            });
+            
+        for (let index = 0; index < sdksToDownload.length; index++) {
+            const element = sdksToDownload[index];
+            // download, extract, cache
+            console.log(tl.loc("InstallingAfresh"));
+            console.log(tl.loc("GettingDownloadUrl", this.packageType, element.name));
+            let downloadUrls = await DotNetCoreReleaseFetcher.getDownloadUrls(osSuffixes, element.name, this.packageType);
+            element.toolPath = await this.downloadAndInstall(downloadUrls);
+            // Prepend the tools path. instructs the agent to prepend for future tasks
+            toolLib.prependPath(element.toolPath);
```

Comment:
This means that the last one will trump all other path entries. And I am not sure whether this will work out if a project targets a version other than this last one.
This task installs each version in a separate folder unlike .Net Core's installer. So, last version's dotnet.exe will not be able to find other versions installed in separate directories.
Have you tried this change by targeting different SDK version in different global.json files. try with version which come from different channel or major version.

Reply:
yes, you are right. I currently test an alternative solution. Please wait 1 day more. I think I'm on a good way.

Reply:
I think I have resolve this. If not reopen this conversion.